cmd/participation/main.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+)
+
+func main() {
+	var osn, channelID, tlsDir, configBlockPath string
+
+	listCmd := flag.NewFlagSet("list", flag.ExitOnError)
+	listCmd.StringVar(&osn, "orderer", "", "Ordering service endpoint")
+	listCmd.StringVar(&tlsDir, "tlsDir", "", "Path to directory containing TLS materials")
+	listCmd.StringVar(&channelID, "channelID", "", "Channel ID")
+
+	removeCmd := flag.NewFlagSet("remove", flag.ExitOnError)
+	removeCmd.StringVar(&osn, "orderer", "", "Ordering service endpoint")
+	removeCmd.StringVar(&channelID, "channelID", "", "Channel ID")
+	removeCmd.StringVar(&tlsDir, "tlsDir", "", "Path to directory containing TLS materials")
+
+	joinCmd := flag.NewFlagSet("join", flag.ExitOnError)
+	joinCmd.StringVar(&osn, "orderer", "", "Ordering service endpoint")
+	joinCmd.StringVar(&tlsDir, "tlsDir", "", "Path to directory containing TLS materials")
+	joinCmd.StringVar(&channelID, "channelID", "", "Channel ID")
+	joinCmd.StringVar(&configBlockPath, "configBlock", "", "Path to file containing config block")
+
+	var (
+		resp *http.Response
+		err  error
+	)
+
+	switch os.Args[1] {
+	case "list":
+		listCmd.Parse(os.Args[2:])
+		if listCmd.NFlag() >= 2 {
+			if channelID != "" {
+				resp, err = listSingleChannel(osn, tlsDir, channelID)
+				break
+			}
+			resp, err = listAllChannels(osn, tlsDir)
+		}
+	case "remove":
+		removeCmd.Parse(os.Args[2:])
+		if removeCmd.NFlag() >= 3 {
+			resp, err = remove(osn, tlsDir, channelID)
+		}
+	case "join":
+		joinCmd.Parse(os.Args[2:])
+		if joinCmd.NFlag() >= 3 {
+			resp, err = join(osn, tlsDir, channelID, configBlockPath)
+		}
+	default:
+		flag.Parse()
+	}
+
+	if err != nil {
+		fmt.Printf("Error: %s\n", err)
+	}
+
+	printResponse(resp, os.Stdout)
+}
+
+func printResponse(resp *http.Response, out io.Writer) {
+	bodyBytes, err := readBodyBytes(resp.Body)
+	if err != nil {
+		log.Fatalf("failed to read http response body: %s", err)
```

Comment:
Not sure this is the "correct" thing to do here but for now I figured if we can't read the response body something has gone very wrong.